### ISSUE TITLE
2020.0b release

### ISSUE THREAD

**Issue by mosra** [mosra/magnum#453]
Opening early mainly to avoid forgetting the tiny but important things to do.

- [ ] Prefix all CMake options with `CORRADE_` / `MAGNUM_` to make things less shitty for subprojects
  - and provide some backwards compatibility to avoid breaking *everything*

**Comment by ttnghia** [comment]
More examples please 😋

**Comment by mosra** [comment]
@ttnghia yeah I know, sorry that I didn't merge your latest one yet :see_no_evil:

**Comment by ttnghia** [comment]
No problem  😄 Once I'm done with my job search (in another month or so), I may have more examples for you 😏

**Comment by mgood7123** [comment]
for

 ⚠️ Assert redirection to default output makes the messages invisible on Android apps (basically need a variant of https://gist.github.com/mosra/566aedd25eadd07927d4f7e385aac931, but for Android apps only)

why dont you just redirect the output at compile time like i do?

HINT:
to add support for alternative output redirection on other platform's such as IOS, simply `#elif defined` the `DEFAULT_STREAM_*` definitions

for example, this redirects to android's `logcat` when compiling for `android`, and `std::cout` and `std::cerr` when compiling for everything else (i am not sure if IOS uses `std::cout` for printing or not)

```C++
#if defined(CORRADE_TARGET_ANDROID)
#include <Corrade/Utility/AndroidLogStreamBuffer.h>

const char * tag = "magnum";

Corrade::Utility::AndroidLogStreamBuffer bufferDebug {Corrade::Utility::AndroidLogStreamBuffer::LogPriority::Debug, tag};
std::ostream androidLogStreamDebug {&bufferDebug};

Corrade::Utility::AndroidLogStreamBuffer bufferWarning {Corrade::Utility::AndroidLogStreamBuffer::LogPriority::Warning, tag};
std::ostream androidLogStreamWarning {&bufferWarning};

Corrade::Utility::AndroidLogStreamBuffer bufferError {Corrade::Utility::AndroidLogStreamBuffer::LogPriority::Error, tag};
std::ostream androidLogStreamError {&bufferError};

#define DEFAULT_STREAM_D &androidLogStreamDebug
#define DEFAULT_STREAM_W &androidLogStreamWarning
#define DEFAULT_STREAM_E &androidLogStreamError
#else
#define DEFAULT_STREAM_D &std::cout
#define DEFAULT_STREAM_W &std::cerr
#define DEFAULT_STREAM_E DEFAULT_STREAM_W
#endif
```

Corrade/Utility/Debug.cpp

https://gist.github.com/dd78eddf7eee46f474bee9d4696d3c4a

**Comment by jonesmz** [comment]
Were there plans to make a release in 2021 still?

I'm not in any particular need of one, I just prefer to use official release when possible instead of some random commit (e.g. the sdl2 fix you helped me with earlier this year)

**Comment by mosra** [comment]
I'd like to make one, yes. The reason why there wasn't any release since last year is that I'm unfortunately still busy on *really huge* features and I'd like to get them into a somewhat presentable state first, to have something to talk about in the release announcement. Those are:

- the low-level Vulkan backend, which is missing SDL/GLFW integration and some shader-related bits
- the new material import APIs, which support custom material properties but neither Assimp nor glTF importers make use of them yet, making it all kinda useless
- the (in-progress) data-oriented scene import (#525), which finally plugs all holes of the legacy import pipeline and allows Vulkan-friendly batch rendering straight from the imported data

Since a release is mainly a marketing booster for the project, announcing a new version with "look, there's a fancy new &lt;thing&gt; but please ignore it, it's not worth talking about yet" would be wasted effort :) I get your point tho, and sorry that it drags on for so long, fortunately Git master is always stable so relying on some random commit isn't a dangerous thing to do.

**Comment by jonesmz** [comment]
Understood!

Thanks for the explanation.

I'm not in a rush, so no big deal to me.

**Comment by EhWhoAmI** [comment]
How much closer are you to a release?

**Comment by mosra** [comment]
Of the tree main tasks above, it's the Vulkan stuff remaining. Plus the thousand tickboxes above.

**Comment by EpsilonKu** [comment]
Happy new year!!! Time to change title, lmao
